hangups/auth.py

```diff
@@ -1,22 +1,23 @@
-"""Simple Google auth supporting second factor.
+""" Google auth using OAuth2.0
 
-This module should not log user credentials.
+This module logs the refresh_token, which can be revoked from
+ https://security.google.com/settings/security/permissions.
+The authorization is shown as "iOS device".
 
 This code may be debugged by invoking it directly:
     python -m hangups.auth
 """
 
-import robobrowser
 import requests
 import logging
 import getpass
 import json
 
-
 logger = logging.getLogger(__name__)
-LOGIN_URL = 'https://accounts.google.com/ServiceLogin'
-SECONDFACTOR_URL = 'https://accounts.google.com/SecondFactor'
 
+OAUTH2_LOGIN_URL='https://accounts.google.com/o/oauth2/auth?scope=https://www.google.com/accounts/OAuthLogin%20https://www.googleapis.com/auth/userinfo.email&redirect_uri=urn:ietf:wg:oauth:2.0:oob&response_type=code&client_id=936475272427.apps.googleusercontent.com'
+CLIENT_ID = '936475272427.apps.googleusercontent.com'
+CLIENT_SECRET = 'KWsJlkaMn1jGLxQpWxMnOox-'
```

Comment:
Strangely, it says "Android TV" when authenticating but "iOS device" on the account permission page.